### PR TITLE
Card qat 1142 sign up

### PR DESCRIPTION
**Pull request by Elvis-Mieles**
## summary
The critical route of registering as a new user in the system is automated based on the CAR QAT 1142 Sign Up, acceptance criteria

### CODE REVIEW

File: src/test/java/training/sign_up/SignUp.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        reportBuilder.generateReports();
+    }
+
+    @Test
+    public void testParallel() {
+        String karateOutputPath = "target/surefire-reports";
+        Results results = Runner.parallel(getClass(), 5, karateOutputPath);
+        generateReport(results.getReportDir());
+        Assertions.assertEquals(0, results.getFailCount());
+    }
```
Comment (jmzagowolox) [inline src/test/java/training/sign_up/SignUp.java]: don't generate a report in every runner that you create. This code and this report must be in a general runner which execute all the features (in the future)

File: src/test/java/karate-config.js [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
+    	}
   }
   if (env == 'dev') {
-    // customize
-    // e.g. config.foo = 'bar';
+
   } else if (env == 'e2e') {
-    // customize
+
   }
```
Comment (jmzagowolox) [inline src/test/java/karate-config.js]: delete this logic if you won't use

File: src/test/java/karate-config.js [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
   var config = {
     env: env,
-	myVarName: 'someValue'
+    someUrl: 'https://angular-conduit-node.herokuapp.com/api',
+    register:{
+    		username: 'QAtomuyuaiut',
+            email: 'qaua65iony@gmail.com',
+            password: 'elvism67898'
+    	}
   }
```
Comment (jmzagowolox) [inline src/test/java/karate-config.js]: good work with that! BUT we must not commit this data to the repository. Try to move this to a _.properties_ or _.env_ or whatever and don't push it.

File: src/test/java/training/sign_up/json/successfulregistrationresponse.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "user": {
+    "username": "#string",
+    "email": "#string",
+    "token": "#string"
+  }
+}
```
Comment (jmzagowolox) [inline src/test/java/training/sign_up/json/successfulregistrationresponse.json]: please add a space line to the end of the file.

File: src/test/java/training/sign_up/sign_up.feature [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    * def email = register.email
+    * def password = register.password
+    * def register = { user: { username: '#(username)', email: '#(email)', password: '#(password)'} }
+
+    * match  register.user contains {"username": '#notnull'}
+    * match  register.user contains {"email": '#notnull'}
+    * match  register.user contains {"password": '#notnull'}
+
+    * assert  register.user.password.length >= 8
+    * assert /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/.test(register.user.email)
```
Comment (jmzagowolox) [inline src/test/java/training/sign_up/sign_up.feature]: why do you validate this data here? this data will be set by the environment, you don't need to validate it.

File: src/test/java/training/sign_up/sign_up.feature [hunk 6]
```diff
@@ -0,0 +1,25 @@
+Feature: Register a new user
+
+  Background:
+    * url someUrl
+
+  Scenario: Register successfully
+
+    * def username = register.username
+    * def email = register.email
+    * def password = register.password
```
Comment (jmzagowolox) [inline src/test/java/training/sign_up/sign_up.feature]: think about in the running of the tests. If you set a specific user to run this feature, you will need to modify the karate-config file to run the test every time. Try to generate a random mail in order to won't have this problem.

**Review by jmzagowolox — CHANGES_REQUESTED** [review]

**Comment by jmzagowolox** [comment]
good work! In the follow branches don't put the card on the name of the branch, that's not necessary :).

File: src/test/java/training/sign_up/sign_up.feature [hunk 7]
```diff
@@ -0,0 +1,25 @@
+Feature: Register a new user
+
+  Background:
+    * url someUrl
+
+  Scenario: Register successfully
```
Comment (jmzagowolox) [inline src/test/java/training/sign_up/sign_up.feature:13]: I recommend you use a Outline Scenario for this feature to validate all the tests cases :).

**Review by jmzagowolox — CHANGES_REQUESTED** [review]

**Comment by Elvis-Mieles** [comment]
## summary
The corrections listed above are made.

It is eliminated from the report to apply it when the tests are finished automating.

**Comment by Elvis-Mieles** [comment]
##summary

The .properties is changed to a .json and the Sign Up scenarios are reduced

**Comment by Elvis-Mieles** [comment]
## summary

the karate-properties.json file is modified

**Comment by Elvis-Mieles** [comment]
## summary
Asserts are added to the feature sing up scenarios

**Comment by Elvis-Mieles** [comment]
## summary
the response json validation is performed in the UnitTest scenario

**Review by jmzagowolox — APPROVED** [review]
Great job! 🙌

**Pull request merged**